cmd/controller/controller.go

```diff
@@ -99,11 +106,212 @@ func NewController(store *store.AuthStore) *Controller {
+	// clusterRefConsumers := map[string]map[string][]v1a1.Subject{}
+	// for _, crc := range crcList.Items {
+	// 	for _, ref := range crc.References {
+	// 		origin := fmt.Sprint("%s/%s", ref.From.Group, ref.From.Resource)
+	// 		target := fmt.Sprint("%s/%s", ref.To.Group, ref.To.Resource)
+	// 		key := fmt.Sprintf("%s;%s;%s", origin, target, ref.For)
+
+	// 		if _, ok := clusterRefConsumers[key]; !ok {
+	// 			clusterRefConsumers[key] = map[string][]v1a1.Subject{
+	// 				"": {},
+	// 			}
+	// 		}
+	// 		if len(crc.ClassNames) > 0 {
+	// 			c.log.Info("Opt in to ClusterReferenceConsumer ClassNames", "ClusterReferenceConsumer", crc.Name)
+	// 			for _, className := range crc.ClassNames {
+	// 				if _, ok := clusterRefConsumers[key][className]; !ok {
+	// 					clusterRefConsumers[key][className] = []v1a1.Subject{}
+	// 				}
+	// 				clusterRefConsumers[key][className] = append(clusterRefConsumers[key][className], crc.Subject)
+	// 			}
+	// 		} else {
+	// 			clusterRefConsumers[key][""] = append(clusterRefConsumers[key][""], crc.Subject)
+	// 		}
+	// 	}
+	// }
+
 	return ctrl.Result{}, nil
 }
 
-func generateQueueKey(fromGroup, fromResource, toGroup, toResource, forReason string) types.NamespacedName {
-	nn := types.NamespacedName{Name: forReason}
-	nn.Namespace = fmt.Sprintf("%s/%s-%s/%s")
-	return nn
+// func generateQueueKey(fromGroup, fromResource, toGroup, toResource, forReason string) types.NamespacedName {
+// 	nn := types.NamespacedName{Name: forReason}
+// 	nn.Namespace = fmt.Sprintf("%s/%s-%s/%s")
+// 	return nn
+// }
+
+type reference struct {
+	Group         string
+	Resource      string
+	FromNamespace string
+	ToNamespace   string
+	Name          string
+}
+
+func (c *Controller) getReferences(list *unstructured.UnstructuredList, path string) ([]reference, error) {
+	refs := []reference{}
+	for _, item := range list.Items {
+		j := jsonpath.New("test")
+		err := j.Parse(fmt.Sprintf("{%s}", path))
+		if err != nil {
+			c.log.Error(err, "error parsing JSON Path")
+			return refs, err
+		}
+		results := new(bytes.Buffer)
+		err = j.Execute(results, item.UnstructuredContent())
+		if err != nil {
+			c.log.Error(err, "error finding results with JSON Path")
+			return refs, err
+		}
+
+		rawRefs := strings.Split(results.String(), " ")
+
+		for _, rr := range rawRefs {
+			jr := map[string]string{}
+			err = json.Unmarshal([]byte(rr), &jr)
+			if err != nil {
+				c.log.Error(err, "error finding results with JSON Path")
+			}
+			group, hasGroup := jr["group"]
+			if !hasGroup {
+				c.log.Info("Missing group in reference", "ref", jr)
+				continue
+			}
+			resource, hasResource := jr["resource"]
+			if !hasResource {
+				kind, hasKind := jr["kind"]
+				if !hasKind {
+					c.log.Info("Missing kind or resource in reference", "ref", jr)
+					continue
+				}
+				gvr, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Group: group, Version: "v1", Kind: kind})
+				resource = gvr.Resource
+			}
```

Comment:
We've moved on from this to instead requiring the RefPattern to specify the group and resource + limit the json path to only pull names that match that group and resource or kind. All of that makes this unnecessary, though it won't necessarily break anything until we try to pair it with object refs that don't include group or resource (IE `secretNames`).

Reply:
Following our offline sync - removing this part and not switching to CEL for the demo.